azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -597,6 +598,39 @@ func resourceArmContainerGroupCreate(d *schema.ResourceData, meta interface{}) e
 	return resourceArmContainerGroupRead(d, meta)
 }
 
+func resourceArmContainerGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Containers.GroupsClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
```

Comment:
where this is an update, can we parse the ID rather than retrieving these from the state:

```
id, err := azure.ParseAzureResourceID(d.Id())
	if err != nil {
		return err
	}
	resGroup := id.ResourceGroup
	name := id.Path["containerGroups"]
```